commands/start.js

```diff
@@ -4,89 +4,66 @@
-const renderWebpackErrors = require('../lib/render-webpack-errors');
 const startServer = require('../lib/start-server');
 const { writeHtml } = require('../lib/html-compiler');
 const { writeCss } = require('../lib/css-compiler');
-const writeWebpackStats = require('../lib/write-webpack-stats');
 const autoCopy = require('../lib/utils/auto-copy');
+const mirrorDir = require('../lib/utils/mirror-dir');
 const logger = require('../lib/logger');
+const {
+  createWebpackConfig,
+  renderWebpackErrors,
+  writeWebpackStats
+} = require('../lib/webpack');
 
-function start(urc) {
-  logger.log(`Starting underreact. ${chalk.yellow('Wait ...')}`);
+module.exports = main;
 
-  const webpackConfig = createWebpackConfig(urc);
-
-  const onFirstCompilation = () => {
-    Promise.all([
-      writeHtml(urc),
-      writeCss(urc),
-      autoCopy.copy({
-        sourceDir: urc.publicDirectory,
-        destDir: urc.outputDirectory
-      })
-    ])
+function main(urc) {
+  logger.log(`Starting underreact. ${chalk.yellow('Wait ...')}`);
+  return (
+    del(urc.outputDirectory, { force: true })
+      // webpack needs to run first as others depend on the assets
+      .then(() => watchWebpack(urc).webpackFirstRun)
+      .then(() =>
+        Promise.all([
+          writeHtml(urc),
+          writeCss(urc),
+          autoCopy.copy({
+            sourceDir: urc.publicDirectory,
+            destDir: urc.outputDirectory
+          })
+        ])
+      )
+      .then(() => startServer(urc))
       .then(() => {
-        // HTML
-        const watcherHtml = chokidar.watch(urc.htmlSource, {
-          ignoreInitial: true
-        });
-        watcherHtml.on('all', () => {
-          logger.log('Writing HTML');
-          writeHtml(urc).catch(logger.error);
-        });
-        watcherHtml.on('error', logger.error);
-
-        // CSS
-        const watchStyleSheets = chokidar.watch(urc.stylesheets, {
-          ignoreInitial: true
-        });
-        watchStyleSheets.on('all', () => {
-          logger.log('Writing CSS');
-          writeCss(urc).catch(logger.error);
-        });
-        watchStyleSheets.on('error', logger.error);
-
-        // Public Dir files
-        const copier = autoCopy.watch({
-          sourceDir: urc.publicDirectory,
-          destDir: urc.outputDirectory
-        });
-        copier.on('copy', filename => {
-          logger.log(`Copying ${filename}`);
-        });
-        copier.on('delete', filename => {
-          logger.log(`Deleting ${filename}`);
-        });
-        copier.on('error', logger.error);
-
-        startServer(urc);
+        watchHtml(urc);
+        watchCss(urc);
+        watchPublicDir(urc);
       })
-      .catch(logger.error);
-  };
-
-  del.sync(urc.outputDirectory, { force: true });
+  );
+}
 
-  const compiler = webpack(webpackConfig);
-  let hasCompiled = false;
+function watchWebpack(urc) {
+  const webpackConfig = createWebpackConfig(urc);
+  const watcher = webpack(webpackConfig);
```

Comment:
The reason I named it watcher was to be consistent with other watch functions, which return watcher and have a `.close` method. But I can make it compiler to avoid confusion.